protoc-gen-kotlin/lib/src/commonMain/kotlin/pbandk/gen/CodeGenerator.kt

```diff
@@ -26,7 +30,35 @@ open class CodeGenerator(
         fn().also { indent = indent.dropLast(4) }
     }
 
-    protected fun writeType(type: File.Type, parentType: String? = null) = when (type) {
+    private fun writeExtension(field: File.Field) {
+        when (field) {
+            is File.Field.Numbered -> {
+                val localType = when (field.type) {
+                    File.Field.Type.MESSAGE -> ((field as File.Field.Numbered.Standard).localTypeName ?: field.kotlinLocalTypeName)?.substringAfter(".")
+                    else -> field.type.standardTypeName
+                }
+                val fieldExtendee = field.extendee!!.trimStart('.').replace("google.protobuf", "pbandk.wkt")
+                line(
+                    "val $fieldExtendee.${field.name}: ${localType}? get() ="
```

Comment:
Everywhere you use `field.name` should be replaced with `field.kotlinFieldName` when you're generating Kotlin code. `field.name` is the field name from the proto file, which might not conform to Kotlin syntax.